usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,743 @@
+
+    var laser = null;
+    
+    var readyToDraw = false;
+    var initialLineStartDataReady = false;
+
+    var PaintSphere = function() {
+        _this = this;
+    };
+
+    PaintSphere.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            Script.setTimeout(function() {
+                var properties = Entities.getEntityProperties(_this.entityID, ['userData','color']);
+                _this.color = properties.color;
+                _this.texture = JSON.parse(properties.userData).textureURL;
+                _this.findWhiteboard();
+                readyToDraw = true;
+            }, WAIT_FOR_ENTITIES_TO_LOAD_MS);
+            dominantHand = MyAvatar.getDominantHand();
+            dominantHandJoint = (dominantHand === "right") ? "RightHand" : "LeftHand";
+            MyAvatar.dominantHandChanged.connect(_this.handChanged);
+
+            parentJointIndex = MyAvatar.getJointIndex(dominantHandJoint + "Index4");
+            if (parentJointIndex === -1) {
+                MyAvatar.getJointIndex(dominantHandJoint + "Index3");
+            }
+            if (parentJointIndex === -1) {
+                MyAvatar.getJointIndex(dominantHandJoint);
+                print("ERROR: Falling back to dominant hand joint as index finger tip could not be found");
+            }
+            tablet.tabletShownChanged.connect(_this.tabletShownChanged);
+            HMD.displayModeChanged.connect(_this.displayModeChanged);
+            Window.domainChanged.connect(_this.domainChanged);
+
+            _this.registerControllerMapping();
+            if (HMD.active) {
+                HMD.closeTablet();
+                _this.setUpHMDMode();
+            } else {
+                _this.setUpDesktopMode();
+            }
+            _this.playSound(OPEN_SOUND, OPEN_SOUND_VOLUME, MyAvatar.position, true, false);
+        },
+
+        /* PLAY A SOUND: Plays the specified sound at the position of the user's Avatar using the volume and playback 
+        mode requested. */
+        playSound: function(sound, volume, position, localOnly, loop){
+            if (sound.downloaded) {
+                if (injector) {
+                    injector.stop();
+                    injector = null;
+                }
+                injector = Audio.playSound(sound, {
+                    position: position,
+                    volume: volume,
+                    localOnly: localOnly,
+                    loop: loop
+                });
+            }
+        },
+
+        findWhiteboard: function() {
+            Entities.findEntities(MyAvatar.position, WHITEBOARD_SEARCH_RADIUS_M).forEach(function(entity) {
+                var properties = Entities.getEntityProperties(entity, ['position', "name"]);
+                if (properties.name && properties.name === "Whiteboard") {
+                    if (whiteboard) {
+                        if (Vec3.distance(properties.position, MyAvatar.position) <
```

Comment:
in what case would this conditional be true?

Reply:
It runs through all whiteboards found within a certain distance. So, for the first whiteboard it finds the variable whiteboard will be changed from null to the id of the first whiteboard. If more than 1 whiteboard was found it will take the next one and compare the distances. The closes one will be stored in the variable.

Reply:
ok - so do you mean the code at lines 164-167 to be outside of the `forEach()` ?

Reply:
yes, after finding the closest whiteboard I make a list of it and it's children to use in finding intersections later

Reply:
i'm a little confused by your reply - as your code is now, your `if (whiteboard)` statement at line 164 is _inside_ the `forEach` loop - it is supposed to be?

Reply:
Oh, yes I see what you mean! Fixed it!